src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -35,17 +35,15 @@ public static VisualStudioInstance GetInstance(string dotNetSdkPath)
                 return null;
             }
 
-            string versionContent = File.ReadAllText(versionPath);
-            Match versionMatch = VersionRegex.Match(versionContent);
+            // Preview versions contain a hyphen after the numeric part of the version. Version.TryParse doesn't accept that.
+            Match versionMatch = VersionRegex.Match(File.ReadAllText(versionPath));
 
-            if (!versionMatch.Success)
-            {
-                return null;
-            }
-
-            if (!int.TryParse(versionMatch.Groups[1].Value, out int major) ||
+            if (!versionMatch.Success ||
```

Comment:
I prefer the original flow here because the failure conditions are much more obvious in separate `if`s:

1. The version regex didn't match.
2. We can't parse a version out.
3. the version doesn't match

Here all together it's hard to see that there are separate failure cases.